geo/src/algorithm/rhumb/mod.rs

```diff
@@ -0,0 +1,157 @@
+        let phi2 = to.y().to_radians();
+        let mut delta_lambda = (to.x() - from.x()).to_radians();
+        // if delta_lambda is over 180° take shorter rhumb line across the anti-meridian:
+        if delta_lambda > pi {
+            delta_lambda = delta_lambda - (two * pi);
+        }
+        if delta_lambda < -pi {
+            delta_lambda = delta_lambda + (two * pi);
+        }
+
+        let delta_psi = ((phi2 / two + pi / four).tan() / (phi1 / two + pi / four).tan()).ln();
+        let delta_phi = phi2 - phi1;
+
+        RhumbCalculations {
+            from: *from,
+            to: *to,
+            phi1,
+            delta_lambda,
+            delta_phi,
+            delta_psi,
+        }
+    }
+
+    fn theta(&self) -> T {
+        self.delta_lambda.atan2(self.delta_psi)
+    }
+
+    fn delta(&self) -> T {
+        let threshold = T::from(10.0e-12).unwrap();
+        let q = if self.delta_psi > threshold {
+            self.delta_phi / self.delta_psi
+        } else {
+            self.phi1.cos()
+        };
+
+        (self.delta_phi * self.delta_phi + q * q * self.delta_lambda * self.delta_lambda).sqrt()
+    }
+
+    fn intermediate(&self, fraction: T) -> Point<T> {
+        let delta = fraction * self.delta();
+        let theta = self.theta();
+        let lambda1 = self.from.x().to_radians();
+        calculate_destination(delta, lambda1, self.phi1, theta)
+    }
+
+    fn intermediate_fill(&self, max_delta: T, include_ends: bool) -> Vec<Point<T>> {
+        let theta = self.theta();
+        let lambda1 = self.from.x().to_radians();
+
+        let total_delta = self.delta();
+
+        if total_delta <= max_delta {
+            return if include_ends {
+                vec![self.from, self.to]
+            } else {
+                vec![]
+            };
+        }
+
+        let number_of_points = (total_delta / max_delta).ceil();
+        let interval = T::one() / number_of_points;
+
+        let mut current_step = interval;
+        let mut points = if include_ends {
+            vec![self.from]
+        } else {
+            vec![]
+        };
+
+        while current_step < T::one() {
+            let delta = total_delta * current_step;
+            let point = calculate_destination(delta, lambda1, self.phi1, theta);
+            points.push(point);
+            current_step = current_step + interval;
+        }
+
+        if include_ends {
+            points.push(self.to);
+        }
+
+        points
+    }
+}
+
+fn calculate_destination<T: CoordFloat + FromPrimitive>(
+    delta: T,
+    lambda1: T,
+    phi1: T,
+    theta: T,
+) -> Point<T> {
+    let pi = T::from(std::f64::consts::PI).unwrap();
+    let two = T::one() + T::one();
+    let four = two + two;
+    let threshold = T::from(10.0e-12).unwrap();
+
+    let delta_phi = delta * theta.cos();
+    let mut phi2 = phi1 + delta_phi;
+
+    // check for some daft bugger going past the pole, normalise latitude if so
```

Comment:
I assume this comment is copied verbatim from the movable-type implementation?

Reply:
Indeed, it was in the movable-type one and Turf kept it, so I did too. Happy to clean it up though if you prefer something more... professional :sweat_smile:.

Reply:
Oh no, was just checking!